addons/im_livechat/static/src/legacy/widgets/public_livechat_view/public_livechat_view.js

```diff
@@ -111,8 +100,7 @@ const PublicLivechatView = Widget.extend({
         // copy so that reverse do not alter order in the thread object
         const messages = _.clone(this.messaging.publicLivechatGlobal.publicLivechat.widget.getMessages());
 
-        const modeOptions = options.isCreateMode ? this._disabledOptions :
-                                                    this._enabledOptions;
+        const modeOptions = this._enabledOptions;
```

Comment:
by the way `this._enabledOptions` does not seem to be used anywhere else now that you removed `this._disabledOptions`; maybe you could directly hardcode its value here and get rid of the variable?

Reply:
I prefer to do it in following PR.
Reason is that it's theoretically less efficient to call `_.defaults()` on each render of thread, rather than store `_.defaults()` once. And also diff can be harder to review and/or debug in case of red runbot for some weird reasons.

Reply:
I wonder if the call to `defaults` could be removed too. There seems to be only one place where `PublicLivechatView` is instantiated, with always the same options. These options could be directly declared here instead, and the defaults removed.

Reply:
Definitely yes, will do this cleaning in a following PR. Thanks!